### PR TITLE
feat(tests,ci): add support for testing from edge snap revision for ci

### PR DESCRIPTION
**Pull request by jat-canonical**
This commit adds support for testing from the edge revision of the AMS snap as the track available for the charm right now corresponds to the snap version for which we only have edge published.

### CODE REVIEW

File: .github/workflows/release.yaml [hunk 1]
```diff
@@ -8,21 +8,33 @@ on:
...
     name: Release charm
     needs:
       - ci
+      - get-version
     uses: canonical/data-platform-workflows/.github/workflows/release_charm.yaml@v13.0.0
     with:
       # TODO: change this when we actually want to cutover the charm to main
       # channels
-      channel: 1.21-ops/edge
+      channel: ${{ jobs.get-version.outputs.version }}-ops/edge
```
Comment (adglkh) [inline .github/workflows/release.yaml]: As you added `get-version` to the release job, why not just use `snap_channel: ${{ needs.get-version.outputs.version }}/edge`  ?

File: .github/workflows/release.yaml [hunk 2]
```diff
@@ -8,21 +8,33 @@ on:
...
+    uses: ./.github/workflows/get-version.yaml
+
   ci:
     name: Build & Test
     uses: ./.github/workflows/build-and-test.yaml
     secrets: inherit
+    needs:
+      - get-version
+    with:
+      snap_channel: ${{ needs.get-version.outputs.version }}/edge
```
Comment (adglkh) [inline .github/workflows/release.yaml]: To me. this should be named `snap_risk_level` instead of `snap_channel`, as the snap channel should always align with what is recorded in the version file while here we would only want to control for managing the risk level of the snap to be installed.

File: .github/workflows/build-and-test.yaml [hunk 3]
```diff
@@ -2,7 +2,13 @@ name: Build & Test
 
 on:
   workflow_call:
-     outputs:
+    inputs:
+      snap_channel:
```
Comment (adglkh) [inline .github/workflows/build-and-test.yaml]: Same below

**Review by adglkh — CHANGES_REQUESTED** [review]

File: .github/workflows/release.yaml (hunk 1 above)
Comment (jat-canonical) [reply .github/workflows/release.yaml]: The version value is picked from `version` file and is plain `1.2X`

File: .github/workflows/release.yaml (hunk 2 above)
Comment (jat-canonical) [reply .github/workflows/release.yaml]: the `version` file is supposed to represent the version of the charm and not necessarily the snap. However, to couple them, is the reason I had only put in the config param for `risk_level` and not version, to be able to change only the risk level and not the snap version being installed. Also in our other repos, the `.base_version` only specifies the version and not the risk level anyway.

**Review by ajanon — APPROVED** [review]
LGTM

File: .github/workflows/release.yaml (hunk 1 above)
Comment (adglkh) [reply .github/workflows/release.yaml]: Sorry, I didn't make it clear. 
What I mean was that since you added
```
    needs:
      - ci
+    - get-version
```
Can this be 
```
channel: ${{ needs.get-version.outputs.version }}-ops/edge
```
for consistency just as you did for the `ci` job or drop the `needs` and use the above version?

File: .github/workflows/release.yaml (hunk 2 above)
Comment (adglkh) [reply .github/workflows/release.yaml]: +1, this aligns with the rationality I mentioned earlier. Currently, the charm installs the snap using the following:
```
self.ams.install(channel=f"{CHARM_VERSION}/{snap_risk_level}", revision=revision)
```

The CHARM_VERSION is obtained from the version file, which is handled by the charm itself. Therefore, I fail to see the necessity of reading the version file in the `get-version` workflow. This leads me to believe that we only require the `snap_risk_level` here. If we agree on this, when passing the parameter to the `build-and-test` workflow, the following `cut` command is unnecessary:
```
if [ -n "${{ inputs.snap_channel }}" ]; then
    risk_level="$(echo ${{ inputs.snap_channel }} | cut -d'/' -f2)"
    args="${args} --snap-risk-level ${risk_level}"
fi
```

That would make the workflow simpler.

File: .github/workflows/release.yaml [hunk 4]
```diff
@@ -8,21 +8,31 @@ on:
...
 jobs:
+  get-version:
+    uses: ./.github/workflows/get-version.yaml
+
   ci:
     name: Build & Test
     uses: ./.github/workflows/build-and-test.yaml
     secrets: inherit
+    with:
+      snap_risk_level: edge
```
Comment (adglkh) [inline .github/workflows/release.yaml:26]: making the `snap_risk_level` configurable as an input would be much better for the release workflow.
(we can do that in a separate PR.)

**Review by adglkh — APPROVED** [review]
LGTM. thanks!
